### ISSUE TITLE
Helm install Error

### ISSUE THREAD

**Issue by B1F030** [kubernetes-sigs/kueue#1459]
**What happened**:
```
[root@test-1 charts]# helm install kueue kueue/ --create-namespace --namespace kueue-system
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest:
 error validating "": error validating data: [apiVersion not set, kind not set]
```
**What you expected to happen**:
install success
**How to reproduce it (as minimally and precisely as possible)**:
```
git clone https://github.com/kubernetes-sigs/kueue.git
cd kueue/charts
helm install kueue kueue/ --create-namespace --namespace kueue-system
```
**Anything else we need to know?**:
I tried `helm template ./charts/kueue --values ./charts/kueue/values.yaml` to debug where did this error happen,
then I find out:
```
---
# Source: kueue/templates/visibility/kustomization.yaml
resources:
- apiservice.yaml
- role_binding.yaml
- service.yaml
---
```
This is where helm got Error, so I tried to remove `kueue/templates/visibility/kustomization.yaml`, it works.
So I wonder, should we remove this file?

**Environment**:
- Kubernetes version (use `kubectl version`): v1.26.0
- Kueue version (use `git describe --tags --dirty --always`): main
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools: Helm
- Others: helm version: v3.9.0

**Comment by B1F030** [comment]
I can help fix that, just need to make sure if that is feasible first.
/assign

**Comment by B1F030** [comment]
@PBundyra What do you think?

**Comment by B1F030** [comment]
I found that this bug may not be easily fixed by removing the kustomization.yaml.
Maybe we need to keep the file, but still have to fix this bug.
I tried add apiVersion and kind for it:
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  - apiservice.yaml
  - role_binding.yaml
  - service.yaml
```
But it still don't work:
```
[root@test-1 charts]# helm install kueue kueue/ --namespace kueue-system
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest:
 resource mapping not found for name: "" namespace: "" from "": 
no matches for kind "Kustomization" in version "kustomize.config.k8s.io/v1beta1"
```
I'll keep trying to fix this.

**Comment by tenzen-y** [comment]
@B1F030 We can remove the `kueue/templates/visibility/kustomization.yaml` file since helm doesn't recognize customization files.

**Comment by tenzen-y** [comment]
cc @mimowo

**Comment by mimowo** [comment]
Here is where we copy the visibility files for helm: https://github.com/kubernetes-sigs/kueue/blob/a68c95860549a85061ae53db73d32de4df3c3446/hack/update-helm.sh#L99-L120. I think we should omit this file.

**Comment by tenzen-y** [comment]
> Here is where we copy the visibility files for helm:
> 
> https://github.com/kubernetes-sigs/kueue/blob/a68c95860549a85061ae53db73d32de4df3c3446/hack/update-helm.sh#L99-L120
> 
> . I think we should omit this file.

Yes, that's right.

**Comment by B1F030** [comment]
If we simply remove the `kustomization.yaml`, there will be new problems:
```
[root@test-1 charts]# kubectl get resourceflavor
E1215 18:17:09.715081   24954 memcache.go:255] couldn't get resource list for visibility.kueue.x-k8s.io/v1alpha1:
 the server is currently unable to handle the request
E1215 18:17:09.723755   24954 memcache.go:106] couldn't get resource list for visibility.kueue.x-k8s.io/v1alpha1:
 the server is currently unable to handle the request
No resources found
```
Is this another bug?

**Comment by mimowo** [comment]
Are you asking about `config/components/visibility/kustomization.yaml`, or  `charts/kueue/templates/visibility/kustomization.yaml`?

I think we should remove it from `charts/kueue/templates/visibility/kustomization.yaml`, and make sure it does not get copied during `./hack/update-helm.sh`. Is there an issue in that case?

**Comment by B1F030** [comment]
> Are you asking about `config/components/visibility/kustomization.yaml`, or `charts/kueue/templates/visibility/kustomization.yaml`?

charts/kueue/templates/visibility/kustomization.yaml

**Comment by mimowo** [comment]
Oh, in that case I'm not sure, it requires investigation.

**Comment by tenzen-y** [comment]
@B1F030 Which kubernetes version do you use? You're using v1.26.0 the same as kubectl version?

**Comment by B1F030** [comment]
> @B1F030 Which kubernetes version do you use? You're using v1.26.0 the same as kubectl version?

yes

**Comment by tenzen-y** [comment]
@B1F030 Recently, we fixed some permission errors related to the visibility server. So, could you verify if this error still happens?

**Comment by B1F030** [comment]
> @B1F030 Recently, we fixed some permission errors related to the visibility server. So, could you verify if this error still happens?

I repulled the git repositry, and try the steps again:
```
git clone https://github.com/kubernetes-sigs/kueue.git
cd kueue/charts
helm install kueue kueue/ --namespace kueue-system
```
still the same problem...
```
[root@test-1 charts]# helm install kueue kueue/ --namespace kueue-system
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: 
error validating "": error validating data: [apiVersion not set, kind not set]
```
The file `kueue/templates/visibility/kustomization.yaml` still exists.

**Comment by tenzen-y** [comment]
> The file `kueue/templates/visibility/kustomization.yaml` still exists.

@B1F030 Yes, we haven't removed that file yet. I would like to ask if https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857626239 still happens.

**Comment by B1F030** [comment]
> @B1F030 Yes, we haven't removed that file yet. I would like to ask if [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857626239) still happens.

Still the same. If I remove that kustomization.yaml and helm install, Errors still happen.

**Comment by tenzen-y** [comment]
> > @B1F030 Yes, we haven't removed that file yet. I would like to ask if [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857626239) still happens.
> 
> Still the same. If I remove that kustomization.yaml and helm install, Errors still happen.

That error happens only for helm charts? Have you seen the same error when using all-in-one installation manifests (`kubectl apply --server-side -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"`) as well?

**Comment by B1F030** [comment]
> That error happens only for helm charts? Have you seen the same error when using all-in-one installation manifests (`kubectl apply --server-side -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"`) as well?

both.

**Comment by tenzen-y** [comment]
I could reproduce the above error in v1.26.3 cluster. Also, that error doesn't happen in v1.27.3 cluster.
If we disable VisibilityServer feature-gate in the v1.26 cluster, https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857626239 seems to happen.

**Comment by tenzen-y** [comment]
@B1F030 Should we work only on https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857440067 error in this issue? And then, should we work on https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857626239 in a separate issue?

**Comment by B1F030** [comment]
> @B1F030 Should we work only on [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857440067) error in this issue? And then, should we work on [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857626239) in a separate issue?

SGTM

**Comment by tenzen-y** [comment]
@B1F030 So, can you create a PR to fix https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857440067 error?

**Comment by tenzen-y** [comment]
> @B1F030 So, can you create a PR to fix [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857440067) error?

I updated the above comment since I put the incorrect link :(

**Comment by B1F030** [comment]
> @B1F030 So, can you create a PR to fix [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857440067) error?

This part could be fixed by simply remove the file `kueue/templates/visibility/kustomization.yaml`, right?

**Comment by tenzen-y** [comment]
> > @B1F030 So, can you create a PR to fix [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857440067) error?
> 
> This part could be fixed by simply remove the file `kueue/templates/visibility/kustomization.yaml`, right?

Also, we should improve scripts to generate helm charts:

https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857465076

**Comment by B1F030** [comment]
> Also, we should improve scripts to generate helm charts:
> 
> [#1459 (comment)](https://github.com/kubernetes-sigs/kueue/issues/1459#issuecomment-1857465076)

Got it. I'm glad to help.
